dbms/src/Interpreters/Join.cpp

```diff
@@ -1140,8 +1165,15 @@ void NO_INLINE joinBlockImplTypeCase(
 
     size_t segment_size = map.getSegmentSize();
     const auto & shuffle_hash_data = shuffle_hash.getData();
-    for (size_t i = 0; i < rows; ++i)
+    assert(probe_process_info.start_row < rows);
+    size_t i;
+    for (i = probe_process_info.start_row; i < rows; ++i)
     {
+        if (probe_process_info.block_full)
+        {
+            probe_process_info.block_full = false;
```

Comment:
a little confusing here, why just set `block_full ` to false here?

Reply:
> a little confusing here, why just set `block_full ` to false here?

And break the loop, maybe I should move it before the loop.

Reply:
done.